src/Aspire.Hosting/Orchestrator/ApplicationOrchestrator.cs

```diff
@@ -140,11 +145,61 @@ static Task PublishUpdateAsync(ResourceNotificationService notificationService,
         }
     }
 
-    private async Task OnResourcesPrepared(OnResourcesPreparedContext _)
+    private async Task OnResourcesPrepared(OnResourcesPreparedContext context)
     {
         await PublishResourcesWithInitialStateAsync().ConfigureAwait(false);
     }
 
+    private Task ProcessResourcesWithoutLifetime(AfterEndpointsAllocatedEvent @event, CancellationToken cancellationToken)
+    {
+        async Task ProcessValueAsync(IResource resource, IValueProvider vp)
+        {
+            try
+            {
+                var value = await vp.GetValueAsync(default).ConfigureAwait(false);
+
+                await _notificationService.PublishUpdateAsync(resource, s =>
+                {
+                    return s with
+                    {
+                        Properties = [
+                            ..s.Properties,
+                            new("Value", value) { IsSensitive = resource is ParameterResource p && p.Secret }
+                        ]
```

Comment:
Are we guaranteed this is the first time we are setting the "Value" property? If it is already there, won't we be adding 2 "Value" properties this way?

Reply:
Fixed